lib/rack/test.rb

```diff
@@ -240,6 +240,7 @@ def env_for(path, env)
       def process_request(uri, env)
         uri = URI.parse(uri)
         uri.host ||= @default_host
+        uri.scheme ||= "https" if env["HTTPS"] == "on"
```

Comment:
OK, if we have used "on" previously I'm fine with that. Thanks for the input @neilang